addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -400,10 +400,15 @@
                             if (successPage.charAt(0) === "#") {
                                 const successAnchorEl = document.getElementById(successPage.substring(1));
                                 if (successAnchorEl) {
-                                    await dom.scrollTo(successAnchorEl, {
-                                        duration: 500,
-                                        extraOffset: 0,
-                                    });
+                                    // Check if the target of the link is a modal.
+                                    if (successAnchorEl.classList.contains("modal")) {
+                                        $(successAnchorEl).modal("show");
```

Comment:
@bvr-odoo Not possible to trigger the hashchange here instead (just making `window.location.href = modalEl.id` ?). That way we'll have the modal opening go though the widget code which would be better :)

Reply:
> window.location.href =

Good idea :+1: I made the change